lorawan-encoding/src/maccommands.rs

```diff
@@ -222,21 +115,154 @@ macro_rules! mac_cmds {
                     $uplink
                 }
 
-                /// length of the payload of the mac command.
+                /// Length of payload without the CID.
                 pub const fn len() -> usize {
                     $size
                 }
+
+                /// Reference to the payload.
+                pub fn bytes (&self) -> &[u8]{
+                    self.0
+                }
             }
         )*
+    }
+}
+
+macro_rules! mac_cmds_enum {
```

Comment:
I am not sure if I fully understood this, but it seems beautiful! 😃

Reply:
heh thanks, macros can make it hard to read but basically I pass an enum to a macro and move some of the existing fns to the macro to avoid writing them for every maccommand. Here some of the fns could probably be inlined. Some places the fns require self even when it is not needed bc it needs to have the same signture. This is especially true for zero length commands